### ISSUE TITLE
Selective module resolution doesn't work with scoped packages

### ISSUE THREAD

**Issue by s-panferov** [yarnpkg/yarn#4242]
#4105 cc @kaylieEB

**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**

```json
"resolutions": {
  "@types/react": "15.0.22"
}
```

Result: 

```
Couldn't find any versions for "react" that matches "15.0.22"
```

The error is that yarn thinks that package `@types/react` is a path `@types`/`react`.

**What is the expected behavior?**
Yarn should correctly install the dep.

The source of bug is [this line](https://github.com/yarnpkg/yarn/blob/840acbf7b0e4c1cce13f8aed54db873e6a7f802e/src/resolution-map.js#L51). We cannot simply split by `/` here.

We either have to
1. Improve parsing rules to properly split scoped packages
2. Use different package separator
3. Use some kind of escaping.

**Please mention your node.js, yarn and operating system version.**
yarn@1.0.0-20170823.1414
node@6.11.2
osx 10.12.6

**Comment by s-panferov** [comment]
Started to work on the fix in #4243

**Comment by kaylieEB** [comment]
@s-panferov awesome! Thanks for working on this :) let me know if you need any help.
